web/app/scripts/workspace/workspace-drawing-board.js

```diff
@@ -684,25 +684,35 @@ angular.module('biggraph').directive(
             op = 'Import ORC';
           }
           const box = scope.workspace.addBox(op, event, { willSaveLater: true });
+          box.parameters.filename = 'uploading...';
           if (op === 'Import CSV') {
             box.parameters.infer = 'yes';
           }
-          uploadFile(file).then(function(filename) {
-            box.parameters.filename = filename;
+          uploadFile(file, function(percent) {
+            const b = scope.workspace.boxMap[box.id];
```

Comment:
Now the whole `box` object survives thanks to the closure. (It's just not the right box object after a reload, if I understand this correctly.) But in this new implementation all we need is the id as we will do a lookup for the fresh box object. So what about saving the id in a separate variable and only capturing that in the closure?

Reply:
> Now the whole `box` object survives thanks to the closure. (It's just not the right box object after a reload, if I understand this correctly.) But in this new implementation all we need is the id as we will do a lookup for the fresh box object. So what about saving the id in a separate variable and only capturing that in the closure?

I'm not terribly worried about the memory use of that small object during an upload. Should I be?